shapes/Shape.cpp

```diff
@@ -25,6 +25,11 @@ void Shape::setPosition(const QPoint& point)
     position = point;
 }
 
+void Shape::draw(QPainter& qp) const
```

Comment:
I would just leave it as pure virtual and call ``applyProperties`` within the derived function's ``draw()``.

Reply:
nah, that makes it hard to forget

Reply:
But you call it anyway ``Shape::draw(qp);`` in each shape